### ISSUE TITLE
Migrations: detect changes in the column

### ISSUE THREAD

**Issue by Dzieni** [StratoKit/strato-db#12]
https://github.com/StratoKit/strato-db/blob/c5e7d77c698a052daf6bc1d5ba73913a8ca6dee5/src/JsonModel/makeMigrations.js#L49-L72

According to this code, we set the migration at all times, even if it's not needed (so if we don't set an index or if we don't make it a real column).

- [ ] don't create empty migrations
- [ ] detect if something has changed (i.e. index has been added) in column settings and add new migrations accordingly

**Comment by wmertens** [comment]
Right, I was always considering to make DB treat table definitions as declarations and automatically update the tables to match, without migrations.

However, this isn't great either because some migrations might depend on column definitions.

Maybe we should make the columns versioned and in the migrations specify the version required? If we keep older column definitions we can even make automatic migrations between column types etc.
